src/visualization/plot_recipes.jl

```diff
@@ -235,6 +260,31 @@ getmesh(pd::PlotData2D) = PlotMesh2D(pd)
 end
 
 
+@recipe function f(pds::PlotDataSeries2D{<:PlotData2D{<:Any, <:AbstractVector{<:AbstractVector}}}) 
```

Comment:
Wow, this is an extremely ugly and obscure dispatch! Since the final word on visualizing curved meshes has not been spoken yet (also, Andrew's unstructured curved mesh is incoming), I think it is OK for now to leave it as it is. However, please create an issue for improving the visualization for curved meshes and also mention it in the description of #397.

Reply:
Please also add some documentation in form of an explanatory comment.

Reply:
We created the issue (#518).